File: restring/src/main/java/com/ice/restring/TextViewTransformer.java
package com.ice.restring;

import android.content.res.Resources;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

/**
 * A transformer which transforms TextView(or any view extends it like Button, EditText, ...):
 * it transforms "text" & "hint" attributes.
 */
class TextViewTransformer implements ViewTransformerManager.Transformer {

    private static final String ATTRIBUTE_TEXT = "text";
    private static final String ATTRIBUTE_ANDROID_TEXT = "android:text";
    private static final String ATTRIBUTE_HINT = "hint";
    private static final String ATTRIBUTE_ANDROID_HINT = "android:hint";

    @Override
    public Class<? extends View> getViewType() {
        return TextView.class;
    }

    @Override
    public View transform(View view, AttributeSet attrs) {
        if (view == null || !getViewType().isInstance(view)) {
            return view;
        }
        Resources resources = view.getContext().getResources();

        for (int index = 0; index < attrs.getAttributeCount(); index++) {
            String attributeName = attrs.getAttributeName(index);
            switch (attributeName) {
                case ATTRIBUTE_ANDROID_TEXT:
                case ATTRIBUTE_TEXT: {
                    String value = attrs.getAttributeValue(index);
                    if (value != null && value.startsWith("@")) {
                        setTextForView(view, resources.getString(attrs.getAttributeResourceValue(index, 0)));
                    }
                    break;
                }
                case ATTRIBUTE_ANDROID_HINT:
                case ATTRIBUTE_HINT: {
                    String value = attrs.getAttributeValue(index);
                    if (value != null && value.startsWith("@")) {
                        setHintForView(view, resources.getString(attrs.getAttributeResourceValue(index, 0)));
                    }
                    break;
                }
            }
        }
        return view;
    }

    private void setTextForView(View view, String text) {
        ((TextView) view).setText(text);
    }

    private void setHintForView(View view, String text) {
        ((TextView) view).setHint(text);
    }
}

File: restring/src/main/java/com/b3nedikt/restring/TextViewTransformer.kt
package com.b3nedikt.restring

import android.util.AttributeSet
import android.view.View
import android.widget.TextView

/**
 * A transformer which transforms TextView(or any view extends it like Button, EditText, ...):
 * it transforms "text" & "hint" attributes.
 */
internal class TextViewTransformer : ViewTransformerManager.Transformer {

    override val viewType = TextView::class.java

    override fun transform(view: View, attrs: AttributeSet): View {
        if (!viewType.isInstance(view)) {
            return view
        }
        val resources = view.context.resources

        for (index in 0 until attrs.attributeCount) {
            val attributeName = attrs.getAttributeName(index)
            when (attributeName) {
                ATTRIBUTE_ANDROID_TEXT, ATTRIBUTE_TEXT -> {
                    val value = attrs.getAttributeValue(index)
                    if (value != null && value.startsWith("@")) {
                        setTextForView(view, resources.getString(attrs.getAttributeResourceValue(index, 0)))
                    }
                }
                ATTRIBUTE_ANDROID_HINT, ATTRIBUTE_HINT -> {
                    val value = attrs.getAttributeValue(index)
                    if (value != null && value.startsWith("@")) {
                        setHintForView(view, resources.getString(attrs.getAttributeResourceValue(index, 0)))
                    }
                }
            }
        }
        return view
    }

    private fun setTextForView(view: View, text: String) {
        (view as TextView).text = text
    }

    private fun setHintForView(view: View, text: String) {
        (view as TextView).hint = text
    }

    companion object {

        private const val ATTRIBUTE_TEXT = "text"
        private const val ATTRIBUTE_ANDROID_TEXT = "android:text"
        private const val ATTRIBUTE_HINT = "hint"
        private const val ATTRIBUTE_ANDROID_HINT = "android:hint"
    }
}
